R/calc_mutation_frequency_bin_region.R

```diff
@@ -0,0 +1,299 @@
+#' @title Calculate Mutation Frequency By Sliding Window.
+#'
+#' @description Count the number of mutations in a sliding window across a 
+#' region for all samples.
+#'
+#' @details This function is called to return the mutation frequency for a given
+#' region, either from a provided input maf data frame or from the GAMBL maf data.
+#' Regions are specified with the `region` parameter. Alternatively, the region of
+#' interest can also be specified by calling the function with `chromosome`,
+#' `start_pos`, and `end_pos` parameters. This function operates on a single region. 
+#' To return a matrix of sliding window counts over multiple regions, 
+#' see `calc_mutation_frequency_bin_regions`.
+#'
+#' @param region A string describing a genomic region in the "chrom:start-end" format. 
+#' The region must be specified in this format OR as separate chromosome, start_pos, end_pos arguments.
+#' @param chromosome Chromosome name in region.
+#' @param start_pos Start coordinate of region.
+#' @param end_pos End coordinate of region.
+#' @param these_samples_metadata Optional data frame containing a sample_id column. 
+#' If not providing a maf file, seq_type is also a required column.
+#' @param these_sample_ids Optional vector of sample IDs. Output will be subset 
+#' to IDs present in this vector.
+#' @param this_seq_type Optional vector of seq_types to include in heatmap. 
+#' Default is "genome". Uses default seq_type priority for samples 
+#' with >1 seq_type.
+#' @param maf_data Optional maf data frame. Will be subset to rows where 
+#' Tumor_Sample_Barcode matches provided sample IDs or metadata table. 
+#' If not provided, maf data will be obtained with get_ssm_by_regions().
+#' @param projection Specify which genome build to use. Required. Default grch37.
+#' @param slide_by Slide size for sliding window. Default 100.
+#' @param window_size Size of sliding window. Default 1000.
+#' @param return_format Return format of mutations. Accepted inputs are "long"
+#' and "wide". Long returns a data frame of one sample ID/window per row. 
+#' Wide returns a matrix with one sample ID per row and one window per column. 
+#' Using the "wide" format will retain all samples and windows regardless of 
+#' the drop_unmutated or min_count_per_bin parameters.
+#' @param min_count_per_bin Minimum counts per bin, default is 0. Setting this 
+#' greater than 0 will drop unmutated windows only when return_format is long.
+#' @param return_count Boolean statement to return mutation count per window (TRUE) 
+#' or binary mutated/unmutated status (FALSE). Default is TRUE.
+#' @param drop_unmutated Boolean for whether to drop windows with 0 mutations. 
+#' Only effective with "long" return format.
+#' @param ... Any additional parameters.
+#' 
+#' @return Either a matrix or a long tidy table of counts per window.
+#'
+#' @rawNamespace import(data.table, except = c("last", "first", "between", "transpose"))
+#' @import dplyr tidyr GAMBLR.helpers
```

Comment:
This is a cyclic dependency. GAMBLR.helpers already has GAMBLR.data as a dependency.

Reply:
Agree with :point_up:

Reply:
Right, then we probably need to either relocate the two helper functions that are needed by `calculate_mutation_frequency_bin_region` and `calculate_mutation_frequency_bin_regions` to this package or have them duplicated in this repo. Interested to hear your thoughts on what you think is the best strategy for this moving forward. As, a third alternative the helper functions could be hardcoded inside the functions as well. All options have their own ups and downsides.

Reply:
Yep, for sure - each option has 👍 and 👎 .
I think the `region_to_chunks` is pretty trivial and can be hard-coded here for the purpose of this functionality (can probably also be reduced in the amount of lines/optimized). The `process_regions` is more involving. It already relies on GAMBLR.data and does not seem to be GSC-specific. What about rehoming that function in this package?

Reply:
I like what you are suggesting, let me make these adjustments and do some tests in a new commit to this PR. Thanks!